nixos/modules/services/databases/postgresql.nix

```diff
@@ -208,7 +208,7 @@ in
           ''
             # Create data directory.
             if ! test -e ${cfg.dataDir}/PG_VERSION; then
-              mkdir -m 0700 -p ${cfg.dataDir}
+              mkdir -m 0750 -p ${cfg.dataDir}
```

Comment:
This does not affect existing systems though. I would also add a chown without recursion below the if block.

Reply:
Sadly postgresql doesn't like it: `FATAL:  data directory "/var/db/postgresql-10.0" has group or world access`

Reply:
You could probably work around with posix acls, but this is not something we can do in general.

Reply:
second try on this: https://github.com/NixOS/nixpkgs/pull/65245